artifacts/debian/debian.go

```diff
@@ -0,0 +1,117 @@
+package debian
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/architecture"
+	"kubevirt.io/containerdisks/pkg/docs"
+	"kubevirt.io/containerdisks/pkg/hashsum"
+	"kubevirt.io/containerdisks/pkg/http"
+	"kubevirt.io/containerdisks/pkg/tests"
+)
+
+type debian struct {
+	Arch            string
+	Version         string
+	VersionName     string
+	getter          http.Getter
+	ExampleUserData *docs.UserData
+	envVariables    map[string]string
+}
+
+const (
+	baseURLFmt   = "https://cloud.debian.org/images/cloud/%s/latest/"
+	baseQcow2Fmt = "debian-%s-genericcloud-%s.qcow2"
+	description  = `Debian Generic Cloud images for KubeVirt.
+<br />
+<br />
+Visit [debian.org](https://cloud.debian.org/images/cloud/) to learn more about Debian project.`
+)
+
+func (d *debian) Metadata() *api.Metadata {
+	metadata := &api.Metadata{
+		Name:            "debian",
+		Version:         d.Version,
+		Description:     description,
+		ExampleUserData: *d.ExampleUserData,
+	}
+
+	if d.ExampleUserData != nil {
+		metadata.ExampleUserData = *d.ExampleUserData
+	}
+
+	return metadata
+}
+
+func (d *debian) Inspect() (*api.ArtifactDetails, error) {
+	var baseURL string
+	var qCowImageName string
+	if strings.HasPrefix(d.Version, "11") || strings.HasSuffix(d.Version, "12") {
+		baseURL = fmt.Sprintf(baseURLFmt, d.VersionName)
+		qCowImageName = fmt.Sprintf(baseQcow2Fmt, d.Version, architecture.GetImageArchitecture(d.Arch))
+	} else if strings.HasPrefix(d.Version, "13") {
```

Comment:
IMO we should not ship 13 yet. Saves us this workaround?

Reply:
Besides the fact that 13 is not "stable" yet, we still want to have it available.

Reply:
ACK

Reply:
Agree, it might be a better next step to think about garbage collection old containerdisk, before adding a daily one.